onnxruntime/core/providers/mkldnn/subgraph/mkl_kernel.h

```diff
@@ -0,0 +1,189 @@
+// Copyright(C) 2019 Intel Corporation
+// Licensed under the MIT License
+
+#pragma once
+#ifdef _WIN32
+#pragma warning(disable : 4244)
+#endif
+
+#include "mkldnn.hpp"
+#include "core/common/cpuid_info.h"
+#include "core/providers/mkldnn/subgraph/subgraph.h"
+
+namespace onnxruntime {
+namespace mkl_dnn {
+
+class MklKernel {
+ public:
+  explicit MklKernel(MklNode& node,
+                     MKLDNNExecutionProvider* provider,
+                     std::shared_ptr<MKLContext> mkl_context) {
+    mkl_context_ = mkl_context;
+    mklnode_ptr_ = std::make_shared<MklNode>(node);
+    provider_ = provider;
+    alloc_ = provider_->GetAllocator(0, OrtMemTypeDefault);
+  }
+  virtual ~MklKernel(){};
+
+  virtual void ReadAttributes(const std::unordered_map<std::string,
+                                                       ONNX_NAMESPACE::AttributeProto>& attributes,
+                              const std::string attributes_prefix = "") {
+    ORT_UNUSED_PARAMETER(attributes);
+    ORT_UNUSED_PARAMETER(attributes_prefix);
+  }
+
+  virtual Status CreatePrimitives(const ONNXRunTimeTensor* input_tensors,
+                                  mkldnn::engine& cpu_engine,
+                                  std::vector<mkldnn::primitive>& net,
+                                  mkldnn::memory::format& src_fmt) = 0;
+
+  virtual void ReorderWeights(const ONNXRunTimeTensor* input_tensors, mkldnn::engine& cpu_engine) {
+    ORT_UNUSED_PARAMETER(input_tensors);
+    ORT_UNUSED_PARAMETER(cpu_engine);
+  }
+
+  virtual Status Compute(const ONNXRunTimeTensor* input_tensors,
```

Comment:
i don't think we should call this Compute() 
it's not doing any computation.

Reply:
Does Submit work?

Reply:
Renamed it to Submit. Here we are submitting data to MKLDNN primitives.

Reply:
I think in a previous discussion we talked about "Bind" ?  binding inputs and outputs?

Reply:
Yes. I missed it. Renamed it to Bind.